libexec/klab-fetch

```diff
@@ -5,12 +5,23 @@ if [ -z "$KLAB_OUT" ]; then
   KLAB_OUT=out
 fi
 
+[[ $# -eq 1 ]] || (echo >&2 "Usage: klab fetch \$URL" && exit 1)
 url=$1
+filename=${url##*/}
 
 mkdir -p "$KLAB_OUT"/log
 wget -c "$url" -P "$KLAB_OUT"/log/
-unzip -qu "$KLAB_OUT/log/${url##*/}" -d "$KLAB_OUT"
-# log the proof hash
-hash=$(basename "$url" '.zip')
-echo $hash > $KLAB_OUT/HEAD
+
+case $url in
+  *.zip)
+    unzip -qu "$KLAB_OUT/log/$filename" -d "$KLAB_OUT"
+    ;;
+  *)
```

Comment:
maybe it's a lil nicer to make this case check on `*.tar.gz` (or whatever zstd produces) and then have a default case that produces an error message?

Reply:
The thing is that tar handles all sorts of archives transparently.

When it doesn’t, it will error out. 

This way we’re backwards and forwards compatible.

Reply:
makes sense :+1: